Scala/Scala.sublime-syntax

```diff
@@ -700,7 +874,61 @@ contexts:
     - match: '(?=\S)'
       set: class-parameter-list
 
+  class-parameter-list-maybe-case:
+    - match: '{{rightarrow}}'
+      fail: case
+    - match: '\('
+      scope: punctuation.section.group.begin.scala
+      push:
+        - match: '\)'
```

Comment:
How about adding meta scopes? Those would be required to reliably support syntax based folding.

```suggestion
        - meta_scope: meta.group.scala
        - match: '\)'
```

Reply:
This is going to be a long battle lol. There's a LOT of these types of groups.